paparazzi/src/main/java/app/cash/paparazzi/internal/Renderer.kt

```diff
@@ -188,12 +189,28 @@ internal class Renderer(
   }
 
   private fun getNativeLibDir(): String {
-    val osName = System.getProperty("os.name").toLowerCase(Locale.US)
+    val osName = System.getProperty("os.name").lowercase(Locale.US)
     val osLabel = when {
       osName.startsWith("windows") -> "win"
       osName.startsWith("mac") -> {
-        val osArch = System.getProperty("os.arch").lowercase(Locale.US)
-        if (osArch.startsWith("x86")) "mac" else "mac-arm"
+        // System.getProperty("os.arch") returns the os of the jre, not necessarily of the platform we are running on,
+        // try /usr/bin/arch to get the actual architecture
+        val osArch = ProcessBuilder("/usr/bin/arch")
```

Comment:
this method call runs on every Paparazzi test method run, so I suspect this change would introduce a slight perf regression due to frequent shelling out.

is it fair to assume from the tagged issue that the current check would be OK if and only if the proper JDK had been installed?

if so, then I'd propose moving this `/usr/bin/arch` logic to `checkInstalledJvm`, changing it to a "valid JDK for OS" check, and guarding that method with a static sentinel (since we shouldn't need to check the installed jvm on every method run either).  thoughts?

Reply:
> is it fair to assume from the tagged issue that the current check would be OK if and only if the proper JDK had been installed?

If you are not running in Rosetta then the original check did work, this pr came about because our work policy locks down what versions of the jdk we can install and they currently don't have the arm version available.

> since we shouldn't need to check the installed jvm on every method run either

That is correct, it should only need to be checked once.

Reply:
Updated to only check once, and it prints a warning if the jre and os architectures don't match